bot/subscription/room.py

```diff
@@ -0,0 +1,59 @@
+from telegram import Update, InlineKeyboardButton, InlineKeyboardMarkup
+from telegram.ext import ContextTypes, ConversationHandler, CallbackQueryHandler
+
+from bot.subscription import inline_keyboard_button, CHECK_ROOMS, end_second_level, NEW_SUBSCRIBE, END
+
+
+async def get_room(update: Update, context: ContextTypes.DEFAULT_TYPE) -> int:
+    context.user_data["rooms"] = context.user_data.get("rooms", ["all"])
+    callback_data = update.callback_query.data
+    if callback_data in ["0", "1", "2", "3", "4"]:
+        if "all" in context.user_data["rooms"]:
+            context.user_data["rooms"].remove("all")
+        if callback_data in context.user_data["rooms"]:
+            context.user_data["rooms"].remove(callback_data)
+        else:
+            context.user_data["rooms"].append(callback_data)
+    elif update.callback_query.data == "all":
+        context.userCHECK_PRICE_data["rooms"] = ["all"]
```

Comment:
если бы это был set, а не список
```suggestion
    rooms = context.user_data["rooms"]
    selected = update.callback_query.data
    if selected in ["0", "1", "2", "3", "4"]:
        rooms.discard("all") 
        rooms.remove(selected) if selected in rooms else rooms.add(selected)
    elif selected == "all":
        rooms = {"all"}
```